crates/services/txpool_v2/src/service.rs

```diff
@@ -284,55 +252,471 @@ where
+                txs_to_remove.push(time_txs_submitted.pop_back().expect("qed").1);
+            }
+        }
+
+        let removed;
+        {
+            let mut pool = self.pool.write();
+            removed = pool.remove_transaction_and_dependents(txs_to_remove);
+        }
+
+        for tx in removed {
+            self.shared_state
+                .tx_status_sender
+                .send_squeezed_out(tx.id(), Error::Removed(RemovedReason::Ttl));
+        }
+    }
+
+    fn process_read(&self, request: ReadPoolRequest) {
+        match request {
+            ReadPoolRequest::GetTxIds {
+                max_txs,
+                response_channel,
+            } => {
+                let tx_ids = {
+                    let pool = self.pool.read();
+                    pool.iter_tx_ids().take(max_txs).copied().collect()
+                };
+                if response_channel.send(tx_ids).is_err() {
+                    tracing::error!(
+                        "Failed to send the result back for `GetTxIds` request"
+                    );
+                }
+            }
+            ReadPoolRequest::GetTxs {
+                tx_ids,
+                response_channel,
+            } => {
+                let txs = {
+                    let pool = self.pool.read();
+                    tx_ids
+                        .into_iter()
+                        .map(|tx_id| pool.find_one(&tx_id))
+                        .collect()
+                };
+                if response_channel.send(txs).is_err() {
+                    tracing::error!(
+                        "Failed to send the result back for `GetTxs` request"
+                    );
+                }
+            }
+        }
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
 pub fn new_service<
+    P2P,
+    BlockImporter,
     PSProvider,
     PSView,
     ConsensusParamsProvider,
     GasPriceProvider,
     WasmChecker,
-    MemoryPool,
 >(
     config: Config,
+    p2p: P2P,
+    block_importer: BlockImporter,
     ps_provider: PSProvider,
     consensus_parameters_provider: ConsensusParamsProvider,
     current_height: BlockHeight,
     gas_price_provider: GasPriceProvider,
     wasm_checker: WasmChecker,
-    memory_pool: MemoryPool,
-) -> Service<PSProvider, ConsensusParamsProvider, GasPriceProvider, WasmChecker, MemoryPool>
+) -> Service<PSView>
 where
-    PSProvider: AtomicView<LatestView = PSView> + Clone + Send + Sync,
+    P2P: P2PSubscriptions<GossipedTransaction = TransactionGossipData>,
+    P2P: P2PRequests + Send + Sync + 'static,
+    PSProvider: AtomicView<LatestView = PSView> + Send + Sync + 'static,
     PSView: TxPoolPersistentStorage,
     ConsensusParamsProvider: ConsensusParametersProvider + Send + Sync,
     GasPriceProvider: GasPriceProviderTrait + Send + Sync,
     WasmChecker: WasmCheckerTrait + Send + Sync,
-    MemoryPool: MemoryPoolTrait + Send + Sync,
+    BlockImporter: BlockImporterTrait + Send + Sync,
```

Comment:
These trait bounds look too restrictive, please check if the `Send + Sync` are needed. Also unsure about `'static`.

Reply:
Static is necessary on the persistent storage because we create an Arc from this. Removed the one from PSProvider.
For the `Sync` and `Send` they were added in `ports.rs` and don't removed from here. I have removed them from here, ty :) They all have `Send` `Sync` and `'static` in `ports.rs` to be usable in arcs and in the future that is defined in `async fn run`.

Changes are available here : https://github.com/FuelLabs/fuel-core/pull/2263/commits/1c345c0bc7799e8e64cece36261f105d896323af